### DISCUSSION TITLE
Statamic control panel

### DISCUSSION THREAD

**Discussion by sunscreem** [spatie/laravel-activitylog#955] [General]
Curious if anyone has manage to get the activity log displaying in a Statamic control panel at all?

**Comment by sunscreem** [comment]
Update on this - I'm building it myself. Biggest hurdle so far is that the package really doesn't play nicely when it's statamic user classes instead of illuminate ones.

**Reply by Gummibeer** [reply]
Hey, I assume that the problem you are facing is our requirement that the causer has to be an instance of `Illuminate\Datebase\Eloquent\Model`!? 🤔
We have reduced it to the lowest possible class - but as we use eloquent relationships it has to be a model. Or is it somehow possible to relate an Eloquent Database Model with a Statamic file-based model? 🤔

I think that the only way you have is to customize the `Activity` model for your package, add a `statamic_user_id` column for example and use some custom logic there. Or even override and rebind the causer resolver logic at all.

But I fear that there's nothing we could do in the package to make it Statamic compatible by default!?

**Reply by sunscreem** [reply]
Hey @Gummibeer. yeah - it doesn't accept Statamic's classes for model's events, which is no problem. I'm not really using those, instead just adding things to the properties field.

The problem I hit up against was with the default `causer`. By default Statamic return from `Auth::user()` as a class of `Statamic\Auth\File\User` which throws an exception in the package. Even when using `->byAnonymous()` it would still crash as it was still seeing a model for the user then not being able to use it.

I upgraded from v3 to v4 which allowed me to use the `CauserResolver` class. I have to fake a user. Its a bit of hack but it's working:

It ends up looking a like this record log-in and log-out:

```php

namespace App\Listeners;

use App\Models\User;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Support\Facades\Auth;
use Spatie\Activitylog\Facades\CauserResolver;

class LogAuthActivity
{
    /**
     * Create the event listener.
     *
     * @return void
     */
    public function __construct()
    {
        //
    }

    /**
     * Handle the event.
     *
     * @param  object  $event
     * @return void
     */
    public function handle($event)
    {
        switch (get_class($event)) {
            case 'Illuminate\Auth\Events\Login':
                $type="Logged in";
            break;
            case 'Illuminate\Auth\Events\Logout':
                $type="Logged out";
            break;
        }


        CauserResolver::setCauser(User::make());

        activity('auth')
        ->byAnonymous()
        ->withProperties([
            'user'=>(Auth::user()->id() ?? null),
        ])
        ->log($type);
    }
}

```

Not ideal but it works. 

If you can see anything there where I've gone wrong I'd appreciate the feedback!
